### PR TITLE
Fix Bug 1492086 - Add ASR Snippets Newsletter template

### PR DESCRIPTION
**Pull request by piatra**
Working demo https://snippets-admin.moz.works/show/7894/

In order to test this 

```javascript
Services.prefs.setStringPref(
"browser.newtab.activity-stream.asrouter.whitelistHosts",
"[\"gist.githubusercontent.com\"]"
);
```

Restart the browser.
Then navigate to `about:newtab?endpoint=https://gist.githubusercontent.com/piatra/70234f08696c0a0509d7ba5568cd830f/raw/f5bdddaa700390532fb70cfc33348d42aef6ba4a/messages.json`

#1st Screen
grafik

#2nd Screen (after clicking the button)
grafik

Content isn't really important, just overall layout and function.

### CODE REVIEW

File: test/unit/asrouter/templates/NewsletterSnippet.test.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    const button = wrapper.find("button.ASRouterButton");
+    assert.equal(button.text(), "Click here");
+    assert.equal(button.prop("className"), "ASRouterButton");
+  });
+  it("should call props.onBlock when CTA button is clicked", () => {
+    const wrapper = mountAndCheckProps({text: "bar"});
+
+    wrapper.instance().onButtonClick();
+
+    assert.calledOnce(onBlockStub);
```
[the remark this answers is not in the document]
Comment (piatra) [reply test/unit/asrouter/templates/NewsletterSnippet.test.jsx]: There are two buttons. The hidden dismiss button that shows up when you hover the snippet  (on the far right) and the button to take you to the next scene.

[the remark this answers is not in the document]
Comment (piatra) [reply test/unit/asrouter/templates/NewsletterSnippet.test.jsx]: And it's rendered from `SnippetBase` not `SnippetNewsletter`

**Comment by piatra** [comment]
@glogiotatidis Can you have a look at [the schema](https://github.com/mozilla/activity-stream/pull/4438/files#diff-1367759b5d095897483e97ba2c133a2f) here is [an example payload](https://gist.githubusercontent.com/piatra/70234f08696c0a0509d7ba5568cd830f/raw/b54014f4a97bd54bd936a2b6aab99303fca48640/messages.json) that I used when developing.

**Comment by sarracini** [comment]
@piatra based on the demo, there should be a "finished" scene as well that says "Check your inbox for the confirmation!". Is that meant to be implemented too?

**Comment by glogiotatidis** [comment]
@piatra 

Questions and notes on payload
 - All the variables that are in the `form` wouldn't work. The service would render those in the top level, outside `form`.  Actually we need all variables to be the same as in the newsletter template for AS to maintain compatibility.
 - I see `privacyLink` inside `privacy_notice`. Snippet service would render all links under the `links` section. Is this OK?

Is there a way to test this before merging?

**Comment by sarracini** [comment]
can you give this a rebase on master? some of the locale stuff is busted on this pr when using the latest central

**Comment by sarracini** [comment]
Changes look good, and both sign up success and sign up fail work nicely. Good to merge if Giorgos approves of schema. Thanks!

**Comment by glogiotatidis** [comment]
A few nits:

- Add block button label template variable that defaults to `Remove this`.
- Block after user has successfully subscribed (see https://bugzilla.mozilla.org/show_bug.cgi?id=1482859)
- On error the sign up button re-appears. Does this follow the current design?
- Don't display the icon on success or error scenes

**Comment by glogiotatidis** [comment]
Also please make sure that you're following the fields from the current template ([link](https://github.com/mozmeao/snippets/blob/master/activity-stream/newsletter-subscribe.html#L6-L25)) and where you do not please file a GH issue against mozmeao/snippets-service to update the service template as well.

**Pull request merged**